src/ConditionalLabel.vue

```diff
@@ -0,0 +1,28 @@
+<template>
+  <div>
+      <label v-if="controlGroup.label && !readonly" class="row my-2">
+          <slot></slot>
+      </label>
+      <span v-else-if="controlGroup.label" class="row my-2">
+          <slot></slot>
```

Comment:
I don't think this is quite right - if controlGroup.label is falsy, we still want to see the slot content, because that includes the dynamic form control. So the enclosing span here should instead be a sibling of the slot (as shown in the bit of pseudo-template i posted before), where the span has `v-if` on `controlGroup.label && readonly` and both are enclosed by a div which has `v-else.

Reply:
I've separated the control into its own slot using named slots.